Feature/ae 1627 laatija history

muep opened the pull request.

muep committed:
AE-1627 Add an audit schema

This should keep better in sync the bits of audit tables that are
exposed in HTTP responses.

muep committed:
AE-1627 Add laatija history API

muep committed:
AE-1627 Add kayttaja history API

muep commented on etp-backend/src/main/clj/solita/etp/schema/audit.clj:
```diff
@@ -0,0 +1,11 @@
+            [solita.etp.schema.common :as common-schema]))
+
+(def Audit
+  {;:event-id common-schema/Key
```
Kannattaisiko ottaa noita pois kommentoituja mukaan, vai siivota pois?

hkorpi replied:
ota vaan pois, lisätään myöhemmin jos tarvitaan - luultavasti ei käyttöä käyttöliittymässä

muep commented on etp-backend/src/main/sql/solita/etp/db/kayttaja.sql:
```diff
@@ -9,3 +9,24 @@ select id, etunimi, sukunimi, email, puhelin, passivoitu, rooli_id as rooli,
+       k.rooli_id as rooli,
+       k.ensitallennus,
+       k.cognito_id as cognitoid,
+       --k.virtu$localid,
```
Tätä virtu-osuutta ei ainakaan laatija tarvitse, mutta ei varmaan muutenkaan ole välttämätön?

hkorpi replied:
ei haittaa, laita vaan tulemaan, hyödyllinen jos tämä tehdään myös käyttäjälle ja olemassa olevat skeemat toimii siltään

hkorpi commented on etp-backend/src/main/sql/solita/etp/db/kayttaja.sql:
```diff
@@ -9,3 +9,24 @@ select id, etunimi, sukunimi, email, puhelin, passivoitu, rooli_id as rooli,
+  audit.kayttaja k
+  join kayttaja modifier on k.modifiedby_id = modifier.id
+where k.id = :id
+order by modifytime;
```
modifytime, event_id

hkorpi replied:
event_id tä on käytetty 2. järjestyksenä näissä

hkorpi replied:
event_id toimisi periaatteessa myös yksinään - modifytime on laitettu tähän siksi että tiedon konversio olisi helpompaa
event_id kuvaa siis sen missä järjestyksessä tiedot on laitettu audit tauluun

muep replied:
Vaihdoin

hkorpi replied:
modifytime ei oikeastaan ole edes 100% luotettava koska sehän perustuu aikaan mutta on meille luultavasti riittävän hyvä

hkorpi commented on etp-backend/src/main/sql/solita/etp/db/laatija.sql:
```diff
@@ -107,3 +107,23 @@ where
+  audit.laatija l
+  join kayttaja modifier on l.modifiedby_id = modifier.id
+where l.id = :id
+order by modifytime;
```
modifytime, event_id

muep replied:
Vaihdettu

muep committed:
AE-1627 Order by modifytime, event_id

muep committed:
AE-1627 Clean up solita.etp.schema.audit

It seems that there would be no immediate use for the fields that
could have been added.

muep committed:
AE-1627 Include virtu items in kayttaja history

The db-row->kayttaja function needed some refactoring to allow
coercing to the larger history schema.

hkorpi reviewed (approved):
:+1:
